Swiften/Elements/RosterPayload.h

```diff
@@ -42,7 +42,16 @@ namespace Swift {
                 version_ = version;
             }
 
+            bool hasMIXAnnotationSupport() const {
```

Comment:
This method name doesn't make it clear what it is for. You probably want to enable requesting MIX annotations for roster requests. How about `hasRequestMIXAnnotations()`? Similar for the following method.